### PR TITLE
[Bootstrap] Fix duplication detection in root block finalization

### PR DESCRIPTION
**Pull request by zhangchiqing**
Fix the duplication detection logic in root block finalization.

### CODE REVIEW

File: cmd/bootstrap/cmd/finalize.go [hunk 1]
```diff
@@ -312,29 +315,31 @@ func readRootBlockVotes() []*hotstuff.Vote {
...
 	nodes := append(internalNodes, partnerNodes...)
 
 	// test for duplicate Addresses
 	addressLookup := make(map[string]struct{})
 	for _, node := range nodes {
 		if _, ok := addressLookup[node.Address]; ok {
-			log.Fatal().Str("address", node.Address).Msg("duplicate node address")
+			return nil, fmt.Errorf("duplicate node address: %v", node.Address)
 		}
+		addressLookup[node.Address] = struct{}{}
```
Comment (zhangchiqing) [inline cmd/bootstrap/cmd/finalize.go:327]: This is the key fix.

File: cmd/bootstrap/cmd/finalize.go [hunk 2]
```diff
@@ -312,29 +315,31 @@ func readRootBlockVotes() []*hotstuff.Vote {
...
 	}
 
 	// test for duplicate node IDs
 	idLookup := make(map[flow.Identifier]struct{})
 	for _, node := range nodes {
 		if _, ok := idLookup[node.NodeID]; ok {
-			log.Fatal().Str("NodeID", node.NodeID.String()).Msg("duplicate node ID")
+			return nil, fmt.Errorf("duplicate node ID: %v", node.NodeID.String())
 		}
+		idLookup[node.NodeID] = struct{}{}
```
Comment (zhangchiqing) [inline cmd/bootstrap/cmd/finalize.go:336]: This is the key fix.

Comment (AlexHentschel) [reply cmd/bootstrap/cmd/finalize.go:336]: 😳 wow. we really forgot this 🤯

**Review by AlexHentschel — APPROVED** [review]
Thanks Leo for finding this.

**Review by jordanschalm — APPROVED** [review]

**Pull request merged**